### ISSUE TITLE
The Yolo neural network is very slow when detecting from a PTZ camera

### ISSUE THREAD

**Issue by kolyan288** [ultralytics/ultralytics#19916]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict, Integrations, Other

### Bug

I took the code from the ultralyrics documentation for opencv, but in this case I detect objects in real time using a PTZ camera, and the predictions work with a long delay.

It is necessary to predict through the model(frame, show = True)

 When I write the model (rtsp, show = True, stream = True), everything works without delay, but I need to implement the code exactly according to the previous version. Need help solving this problem, why is this happening?

### Environment

Ultralytics version - 8.3.40
OS - Linux Mint 21.3
Python version 3.10.14
CPU - Intel Core I5, No GPU
opencv-python version 4.10.0.84
onvif_zeep 0.2.12

yolo checks:

![Image](https://github.com/user-attachments/assets/007b47ab-2834-4f3e-b931-c9247cab2986)

### Minimal Reproducible Example

```python
import cv2
from ultralytics import YOLO
from onvif import ONVIFCamera


ip = 'ip'
username = 'username'
password = 'password'

port_rtsp = 'port_rtsp'
port = 'port

cam = ONVIFCamera(ip, port, username, password, wsdl_dir = r'python-onvif-zeep/wsdl')

media = cam.create_media_service()
ptz = cam.create_ptz_service()

media_profile = media.GetProfiles()[0]

def make_rtsp(ip, port, username, password):
    return f"rtsp_link"

rtsp = make_rtsp(ip, port_rtsp, username, password)

# Load the YOLO model
model = YOLO("yolo11n.pt")

# Open the video file
cap = cv2.VideoCapture(rtsp)



request = ptz.create_type("GetConfigurationOptions")
request.ConfigurationToken = media_profile.PTZConfiguration.token

ptz_configuration_options = ptz.GetConfigurationOptions(request)

relative_move_request = ptz.create_type("RelativeMove")
relative_move_request.ProfileToken = media_profile.token
relative_move_request.Speed = {'PanTilt': {'x': 0, 'y': 0}}


# Loop through the video frames
while cap.isOpened():
    # Read a frame from the video
    success, frame = cap.read()

    if success:
        # Run YOLO inference on the frame
        results = model(frame, show = True)

        # Visualize the results on the frame
        annotated_frame = results[0].plot()

        pan_offset = 0.001
        tilt_offset = 0.001
        zoom_offset = 0

        relative_move_request.Translation = {"PanTilt": {"x": pan_offset, # 0.01 * abs_zoom
                                            "y": tilt_offset}, # 0.01 * abs_zoom
                                "Zoom": zoom_offset}

        relative_move_request.Speed = {'PanTilt': {'x': 0, 'y': 0}}

        ptz.RelativeMove(relative_move_request)

        if cv2.waitKey(1) & 0xFF == ord("q"):
            break
    else:
        # Break the loop if the end of the video is reached
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kolyan288, thank you for reaching out and sharing your detailed issue 🚀! We appreciate the effort you've put into providing the context and code snippet.

For 🐛 Bug Reports like this, we kindly ask for a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us diagnose the problem more effectively. Your code is comprehensive, but please ensure it includes only the minimal components required to replicate the issue. This helps us pinpoint the root cause faster.

In the meantime, here are a few steps to verify and troubleshoot:

1. **Upgrade**: Ensure you are using the latest version of the `ultralytics` package and its dependencies. You can upgrade with:
   ```bash
   pip install -U ultralytics
   ```
   Ensure your environment meets the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

2. **Environment Check**: As you are running on CPU, performance can be slower compared to GPU. If possible, try running the same code on a system with a supported GPU to compare performance.

3. **Ultralytics HUB**: Consider testing your workflow on [Ultralytics HUB](https://hub.ultralytics.com/), which offers an optimized and user-friendly platform to train, deploy, and manage your models.

4. **Verified Environments**: YOLO is optimized for various environments. If you want to test alternative setups, you can try running your code in these verified environments:
   - **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
   - **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/).

For further questions, feel free to join our community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time support.
- [Discourse](https://community.ultralytics.com/) for in-depth discussions.
- [Subreddit](https://reddit.com/r/Ultralytics) to share insights with others.

This is an automated response, but an Ultralytics engineer will assist you soon. Thank you for your patience and for being part of the Ultralytics community 🌟!

**Comment by Y-T-G** [comment]
You can use cv2.imshow instead of show=True

`show=True` adds 300ms delay

https://github.com/ultralytics/ultralytics/blob/5c39f5ed8502e991dcbf6ba5f8f750206563e582/ultralytics/engine/predictor.py#L486

**Comment by kolyan288** [comment]
now my code looks like this
```python
import cv2
from ultralytics import YOLO
from onvif import ONVIFCamera


ip = 'ip'
username = 'username'
password = 'password'

port_rtsp = 'port_rtsp'
port = 'port

cam = ONVIFCamera(ip, port, username, password, wsdl_dir = r'python-onvif-zeep/wsdl')

media = cam.create_media_service()
ptz = cam.create_ptz_service()

media_profile = media.GetProfiles()[0]

def make_rtsp(ip, port, username, password):
    return f"rtsp_link"

rtsp = make_rtsp(ip, port_rtsp, username, password)

# Load the YOLO model
model = YOLO("yolo11n.pt")

# Open the video file
cap = cv2.VideoCapture(rtsp)



request = ptz.create_type("GetConfigurationOptions")
request.ConfigurationToken = media_profile.PTZConfiguration.token

ptz_configuration_options = ptz.GetConfigurationOptions(request)

relative_move_request = ptz.create_type("RelativeMove")
relative_move_request.ProfileToken = media_profile.token
relative_move_request.Speed = {'PanTilt': {'x': 0, 'y': 0}}


# Loop through the video frames
while cap.isOpened():
    # Read a frame from the video
    success, frame = cap.read()

    if success:
        # Run YOLO inference on the frame
        results = model(frame, show = False)
        cv2.imshow('frame', results[0].plot())
        

        pan_offset = 0.001
        tilt_offset = 0.001
        zoom_offset = 0

        relative_move_request.Translation = {"PanTilt": {"x": pan_offset, # 0.01 * abs_zoom
                                            "y": tilt_offset}, # 0.01 * abs_zoom
                                "Zoom": zoom_offset}

        relative_move_request.Speed = {'PanTilt': {'x': 0, 'y': 0}}

        ptz.RelativeMove(relative_move_request)

        if cv2.waitKey(1) & 0xFF == ord("q"):
            break
    else:
        # Break the loop if the end of the video is reached
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()
```
and now the code inside the 'while' loop runs only once and does not display the predicted image.
